src/browser/page.zig

```diff
@@ -593,6 +601,76 @@ pub const Page = struct {
         }
     }
 
+    pub const KeyboardEvent = struct {
+        type: Type,
+        key: []const u8,
+        code: []const u8,
+        alt: bool,
+        ctrl: bool,
+        meta: bool,
+        shift: bool,
+
+        const Type = enum {
+            keydown,
+        };
+    };
+
+    pub fn keyboardEvent(self: *Page, kbe: KeyboardEvent) !void {
+        if (kbe.type != .keydown) {
+            return;
+        }
+
+        const Document = @import("dom/document.zig").Document;
+        const element = (try Document.getActiveElement(@ptrCast(self.window.document), self)) orelse return;
+
+        const event = try parser.keyboardEventCreate();
+        defer parser.keyboardEventDestroy(event);
+        try parser.keyboardEventInit(event, "keydown", .{
+            .bubbles = true,
+            .cancelable = true,
+            .key = kbe.key,
+            .code = kbe.code,
+            .alt = kbe.alt,
+            .ctrl = kbe.ctrl,
+            .meta = kbe.meta,
+            .shift = kbe.shift,
+        });
+        _ = try parser.elementDispatchEvent(element, @ptrCast(event));
+    }
+
+    fn keydownCallback(node: *parser.EventNode, event: *parser.Event) void {
+        const self: *Page = @fieldParentPtr("keydown_event_node", node);
+        self._keydownCallback(event) catch |err| {
+            log.err(.browser, "keydown handler error", .{ .err = err });
+        };
+    }
+
+    fn _keydownCallback(page: *Page, event: *parser.Event) !void {
+        const kbe: *parser.KeyboardEvent = @ptrCast(event);
+        const target = (try parser.eventTarget(event)) orelse return;
+        const node = parser.eventTargetToNode(target);
+        const tag = (try parser.nodeHTMLGetTagType(node)) orelse return;
+        switch (tag) {
+            .input => {
+                const element: *parser.Element = @ptrCast(node);
+                const input_type = (try parser.elementGetAttribute(element, "type")) orelse "text";
+                if (std.mem.eql(u8, input_type, "text")) {
```

Comment:
```suggestion
                const input: *parser.Input= @ptrCast(node);
                const input_type = try parser.inputGetType(input);
                if (std.mem.eql(u8, input_type, "text")) {
```
`elementGetAttribute` is kinda broken in libDom, I think we should avoid using it where possible.

Perhaps you are using it here, because in the test use case the value was set via `elementSetAttribute` which has the same broken behavior.. In which case I think we should at least leave a comment.

Reply:
I made the change in a separate commit..I changed other cases and I added a `buttonGetType`